lib/v3/parser.js

```diff
@@ -90,6 +88,53 @@ class Parser extends EventEmitter {
         this.emit('end');
     }
 
+    static getDefaultOptions() {
+        return {
+            includeSourceLocations: true,
+            defaultMethodVisibility: utils.DEFAULT_VISIBILITY,
+            defaultActionVisibility: utils.DEFAULT_VISIBILITY,
+            features: [...SUPPORTED_FEATURES],
+            componentName: null,
+            eventsEmitted: {},
+            identifiers: {},
+            imports: {},
+            dispatcherConstructorNames: [],
+            dispatcherNames: [],
+        };
+    }
+
+    static normalizeOptions(options) {
+        const defaults = Parser.getDefaultOptions();
+
+        Object.keys(defaults).forEach((optionKey) => {
+            // If the key was not set by the user, apply default value.
+            if (!(optionKey in options)) {
+                options[optionKey] = defaults[optionKey];
+            }
+        });
+    }
+
+    static validateOptions(options) {
+        Parser.normalizeOptions(options);
+
+        // Check the presence and basic format of multiple options
+        for (const key of ['eventsEmitted', 'identifiers', 'imports']) {
+            const hasKey = (key in options);
```

Comment:
Looks like that this method partially duplicated with V2 parser and can be extracted to common place